protocol-designer/src/components/modals/CreateFileWizard/utils.ts

```diff
@@ -124,37 +100,155 @@ export const MOVABLE_TRASH_CUTOUTS = [
   },
 ]
 
+export const getUnoccupiedStagingAreaSlots = (
+  modules: FormState['modules']
+): DeckConfiguration => {
+  let unoccupiedSlots = STANDARD_EMPTY_SLOTS
+  const moduleCutoutIds =
+    modules != null
+      ? Object.values(modules).flatMap(module =>
+          module.type === THERMOCYCLER_MODULE_TYPE
+            ? [`cutout${module.slot}`, 'cutoutA1']
+            : `cutout${module.slot}`
+        )
+      : []
+
+  unoccupiedSlots = unoccupiedSlots.filter(emptySlot => {
+    return !moduleCutoutIds.includes(emptySlot.cutoutId)
+  })
+
+  return unoccupiedSlots
+}
+
+export const getNextAvailableModuleSlot = (
+  modules: FormState['modules'],
+  additionalEquipment: FormState['additionalEquipment']
+): string => {
+  const moduleSlots =
+    modules != null
+      ? Object.values(modules).flatMap(module =>
+          module.type === THERMOCYCLER_MODULE_TYPE
+            ? [module.slot, 'A1']
+            : module.slot
+        )
+      : []
+  const stagingAreas = additionalEquipment.filter(equipment =>
+    equipment.includes('stagingArea')
+  )
+  const stagingAreaCutouts = stagingAreas.map(cutout => cutout.split('_')[1])
+  const hasWasteChute = additionalEquipment.find(equipment =>
+    equipment.includes('wasteChute')
+  )
+  const wasteChuteSlot = hasWasteChute ? [WASTE_CHUTE_CUTOUT as string] : []
```

Comment:
seems that hasWasteChute can be `undefined`.

```suggestion
  const wasteChuteSlot = Boolean(hasWasteChute) ? [WASTE_CHUTE_CUTOUT as string] : []
```